cli/pkg/kctrl/cmd/package/installed/create_or_update.go

```diff
@@ -327,6 +352,48 @@ func (o CreateOrUpdateOptions) update(client kubernetes.Interface, kcClient kccl
 	return nil
 }
 
+func (o *CreateOrUpdateOptions) dropValuesSecret(client kubernetes.Interface) (bool, error) {
+	secretName := o.createdAnnotations.SecretAnnValue()
+	pkgiIdentifier := o.createdAnnotations.PackageAnnValue()
+
+	valuesSecret, err := client.CoreV1().Secrets(o.NamespaceFlags.Name).Get(context.Background(), secretName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			o.statusUI.PrintMessagef("Values secret '%s' not found", secretName)
+			return true, nil
+		}
+		return false, fmt.Errorf("Getting values secret: %s", err.Error())
+	}
+
+	// Do not delete secrets not created by kctrl
+	// TODO: Should we not error out and continue with the rest of the changes?
+	annotations := valuesSecret.GetAnnotations()
+	val, found := annotations[KctrlPkgAnnotation]
+	if !found || val != pkgiIdentifier {
+		// To support older version of Tanzu CLI. To be deprecated
+		val, found = annotations[TanzuPkgAnnotation]
+		if !found || val != pkgiIdentifier {
+			return false, fmt.Errorf("Deleting values secret: Secret was not created by kctrl")
+		}
+	}
+
+	o.statusUI.PrintMessagef("Deleting values secret '%s'", secretName)
+	err = client.CoreV1().Secrets(o.NamespaceFlags.Name).Delete(context.Background(), secretName, metav1.DeleteOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Deleting values secret: %s", err.Error())
+	}
+
+	return true, nil
+}
+
+func (o *CreateOrUpdateOptions) removeValuesSecretReference(pkgi *kcpkgv1alpha1.PackageInstall) {
+	for i, valueRef := range pkgi.Spec.Values {
+		if valueRef.SecretRef.Name == o.createdAnnotations.SecretAnnValue() {
+			pkgi.Spec.Values = append(pkgi.Spec.Values[:i], pkgi.Spec.Values[i+1:]...)
```

Comment:
Maybe it won't happen in this specific scenario, but I think this could panic with `slice bounds out of range` in some situations as the variable `i` is not updated after removing a specific item from the slice, so value of `i` could be greater than the length of the slice.

Reply:
You are correct, I was assuming that there won't be more than one value which matches our naming convention.
To be safer I am gonna return after removing one reference now.
Then we do not run into the funky case.

I am wondering if we should just clobber all the values supplied and make it an empty array / nil value. I thought this would be safer though.